kotlinx-coroutines-core/jvm/src/debug/internal/AgentInstallationType.kt

```diff
@@ -3,9 +3,13 @@ package kotlinx.coroutines.debug.internal
 /**
  * Object used to differentiate between agent installed statically or dynamically.
  * This is done in a separate object so [DebugProbesImpl] can check for static installation
- * without having to depend on [kotlinx.coroutines.debug.AgentPremain], which is not compatible with Android.
+ * without having to depend on [AgentPremain], which is not compatible with Android.
  * Otherwise, access to `AgentPremain.isInstalledStatically` triggers the load of its internal `ClassFileTransformer`
  * that is not available on Android.
+ *
+ * Usage Note: Fleet (Reflection): FleetDebugProbes
+ * Usage Note: Android (Hard Coded, ignored for Leak Detection)
+ * Usage Note: IntelliJ (Suppress KotlinInternalInJava): CoroutineDumpState
```

Comment:
I tried to add small notes on internal APIs which are forcefully used by clients (by reflection or ignoring internal visibility). Ss those clients are 'not to be broken', I think its worth marking the code somehow as "internal, but breached'

Reply:
Ok, makes sense. Though a comment would not stop us: who reads them? Better to mark this as `@PublishedApi` so that this appears in the binary compatibility validator dump. Also, it would help a lot if it were reworded, like "This object must not be renamed or moved, because this would break binary compatibility for the following clients forcefully access it:", and then the list. Even with zero context, this gives a clear and actionable impression.